phpBB/phpbb/config/config.php

```diff
@@ -148,6 +148,29 @@ public function set_atomic($key, $old_value, $new_value, $use_cache = true)
 	}
 
 	/**
+	* Checks configuration option's value only if the new_value matches the
+	* current configuration value and the configuration value does exist.Called
+	*only after set_atomic has been called.
+	*
+	* @param  string $key       The configuration option's name
+	* @param  string $old_value Current configuration value
+	* @param  string $new_value New configuration value
+	* @throws \phpbb\exception\http_exception when configuration value is set and not equal to *new_value.
+	* @return bool              True if the value was changed, false otherwise.
+	*/
+	public function ensure_lock($key, $new_value)
+	{
+		if (isset($this->config[$key]) && $this->config[$key] == $new_value)
+		{
+			return true;
+		}
+		else
```

Comment:
@VSEphpbb  What if its not true? Then we need something to cover that case as well. If its not true then we need to raise an exception since even after calling set_atomic we were not able to get a lock.